crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -813,7 +814,7 @@ module sui::sui_system {
             // If we are upgrading the system state, we need to make sure that the protocol version
             // is also upgraded.
             assert!(old_protocol_version != next_protocol_version, 0);
-            let cur_state: SuiSystemStateInner = dynamic_field::remove(&mut wrapper.id, wrapper.version);
+            let cur_state: SuiSystemStateInnerV2 = dynamic_field::remove(&mut wrapper.id, wrapper.version);
```

Comment:
Fine for this PR, but in general should we structure this with a check to make sure wrapper.version is actually 2 here (or whatever the current version is?)